### DISCUSSION TITLE
Fetch prevalue data from node property

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco.Forms.Issues#1463] [Ideas]
Currently it is possible to use prevalue source to fetch data from a specific node or use current node as root.
However it only list children/descendants.

Sometimes the data are just from a picker (Content Picker og MNTP) or Block List now with more element based content (nested content).

Do we have any options to handle this or eventually implement you own prevalue source to handle this?

Currently I find the prevalue source / data source options a bit limited compared to Contentment package which has a lot of options to configure data/prevalue source and implement your own:
https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/data-list.md

**Comment by bjarnef** [comment]
Another idea.

Could Umbraco Forms has a connector to [Umbraco UI Bulder](https://umbraco.com/products/add-ons/ui-builder/)?

Then if installed it could possible re-use configured repositories in Umbraco UI builder?

![image](https://github.com/umbraco/Umbraco.Forms.Issues/assets/2919859/e565a108-7d61-4059-8038-237811d480d0)

Not a specific customer requirement we have had, more just a thought. I guess in could use entity/repository picker from UI builder, so it didn't need to configure the database options, but the caching option could still exists.

It could be an separate package with dependency on Umbraco Forms and Umbraco UI Builder.

I guess it make also just be a package idea or someone using Umbraco UI Builder as we have used it in any projects yet.

**Comment by Amalie-Wowern** [comment]
Hi @bjarnef 

This is something we already have in our backlog, i will let you know when this is something we will look into
